cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/datastore/DatastoreSchemaTests.java

```diff
@@ -405,4 +406,218 @@ public Integer call(SQLDatabase db) throws Exception {
+
+            assertLeafCount(datastore, "Document15",392);
+            assertAttachmentCount(datastore, "Document15",1127);
+            assertWinner(datastore, "Document15","7-rootcdddbd");
+
+            assertLeafCount(datastore, "Document16",1);
+            assertAttachmentCount(datastore, "Document16",0);
+            assertWinner(datastore, "Document16","1-root");
+
+            assertLeafCount(datastore, "Document17",326);
+            assertAttachmentCount(datastore, "Document17",1051);
+            assertWinner(datastore, "Document17","7-rootcddcbe");
+
+            assertLeafCount(datastore, "Document18",124);
+            assertAttachmentCount(datastore, "Document18",379);
+            assertWinner(datastore, "Document18","7-rootadecbd");
+
+            assertLeafCount(datastore, "Document19",1);
+            assertAttachmentCount(datastore, "Document19",0);
+            assertWinner(datastore, "Document19","1-root");
+
+            assertLeafCount(datastore, "Document20",248);
+            assertAttachmentCount(datastore, "Document20",724);
+            assertWinner(datastore, "Document20","7-rootbecbec");
+
+            assertLeafCount(datastore, "Document21",49);
+            assertAttachmentCount(datastore, "Document21",140);
+            assertWinner(datastore, "Document21","7-rootabaeac");
+
+            assertLeafCount(datastore, "Document22",1);
+            assertAttachmentCount(datastore, "Document22",0);
+            assertWinner(datastore, "Document22","1-root");
+
+            assertLeafCount(datastore, "Document23",258);
+            assertAttachmentCount(datastore, "Document23",805);
+            assertWinner(datastore, "Document23","7-rootbddabd");
+
+            assertLeafCount(datastore, "Document24",840);
+            assertAttachmentCount(datastore, "Document24",2358);
+            assertWinner(datastore, "Document24","7-rootedbebe");
+
+        } finally {
+            datastore.close();
+        }
+    }
+
+    // utility methods used by tests
+    private void assertLeafCount(Datastore datastore, String docId, int expectedLeafCount) {
+        int actualLeafCount = ((DatastoreImpl)datastore).getAllRevisionsOfDocument(docId).leafRevisions().size();
+        Assert.assertEquals(expectedLeafCount, actualLeafCount);
+    }
+
+    private void assertAttachmentCount(Datastore datastore, String docId, int expectedAttachmentCount) {
+        int actualAttachmentCount = 0;
+        Set<String> revIds = ((DatastoreImpl)datastore).getAllRevisionsOfDocument(docId).leafRevisionIds();
+        for (String revId : revIds) {
+            try {
+                actualAttachmentCount += datastore.getDocument(docId, revId).attachments.size();
+            } catch (DocumentNotFoundException dnfe) {
+                Assert.fail("Failed to find revision for document ID " + docId + " and revision ID " + revId);
+            }
+        }
+        Assert.assertEquals(expectedAttachmentCount, actualAttachmentCount);
```

Comment:
Message please see https://github.com/cloudant/sync-android/pull/336#discussion_r75153045 for my opinion on this